props/blaster.h

```diff
@@ -112,39 +113,10 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     return random <= percent ? true : false;
   }
 
-  virtual void Fire() {
-#ifdef ENABLE_MOTION
-#ifdef BLASTER_JAM_PERCENTAGE
-    // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
-    is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
-
-    if (is_jammed_) {
-      SaberBase::DoEffect(EFFECT_JAM, 0);
-      return;
-    }
-#endif
-#endif
-
-    if (max_shots_ != -1) {
-      if (shots_fired_ >= max_shots_) {
-        SaberBase::DoEffect(EFFECT_EMPTY, 0);
-        return;
-      }
-    }
-
-    if (blaster_mode == MODE_AUTO) {
-      SelectAutoFirePair(); // Set up the autofire pairing if the font suits it.
-      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
-      SaberBase::DoBeginLockup();
-      auto_firing_ = true;
-    } else {
-      if (blaster_mode == MODE_STUN) {
-        SaberBase::DoEffect(EFFECT_STUN, 0);
-      } else {
-        SaberBase::DoEffect(EFFECT_FIRE, 0);
-      }
-
-      shots_fired_++;
+  virtual void CheckEmpty() {
```

Comment:
might be helpful if CheckEmpty() returns true/false so you can just do:

if (CheckEmpty()) return;